tools/src/wyvern/target/corewyvernIL/metadata/HasMetadata.java

```diff
@@ -0,0 +1,9 @@
+package wyvern.target.corewyvernIL.metadata;
+
+import java.util.Iterator;
+
+public interface HasMetadata {
+    public Metadata[] getMetadata();
```

Comment:
My thought there was that Set contains operations like clear() and remove(elem), so it would be possible for a caller to delete metadata from an ASTNode/OIRAST.

If that's not a concern, then we could have that return a Set.